src/MedTrackerAPI.Tests/Features/Supplies/AddSupplyTests.cs

```diff
@@ -0,0 +1,60 @@
+using FluentValidation.TestHelper;
+using MedTrackerAPI.Features.Supplies;
+
+namespace MedTrackerAPI.Tests.Features.Supplies;
+
+public class AddSupplyTests
+{
+    [TestCase(null)]
+    [TestCase("")]
+    public void GivenNoDescription_ThenIsNotValid(string? value) =>
+        new AddSupply.AddSupplyCommandValidator().TestValidate(new AddSupply.AddSupplyCommand
+        {
```

Comment:
I’d create what is called a “Fake factory” for this that makes a valid one and then the test just makes it bad it some way. Helps with duplication

Reply:
I'm looking into this.

Reply:
I added a FakeFactory. I'm not sure about my implementation though. I have a couple of questions. Should I not create a device and then create a supply based on that device id, or just create the supplies with a provided value in the setup method?

If I always did the device I could make the CreateFakeSuppliesForDevice method private. The way it's set in the DB you'd never have supplies without a Device.

Reply:
The dependency chain of projects essentially makes it private anyway. It makes sense for the FakeFactory to be public to the tests project. Honestly, there's fewer sins (IMO) that can be committed in test projects.

Reply:
If a supply can't exist without a device, it makes sense to force a CreateFakeDevice() and then you can chain something like .WithSupply(). I accomplish it with something like this ->

```
 public static User CreateFakeUser()
        => Faker.Build<User>()
            .With(f => f.LoadSlipCount, RandomNumber<int>(1))
            .With(f => f.Id, Guid.NewGuid())
            .With(f => f.AddressLocation, new Point(-93.642470, 41.706610) { SRID = GoogleApiSettings.SRID })
            .With(f => f.Email, RandomString() + "@gmail.com")
            .Without(f => f.Insurance)
            .Without(f => f.Subscription)
            .Without(f => f.LoadSlips)
            .Without(f => f.PreferredTruckCapacities)
            .Without(f => f.PreferredTruckTypes)
            .Without(f => f.AuthorizedForUsers)
            .Without(f => f.AuthorizedUsers)
            .With(f => f.UserSetting, CreateFakeUserSetting())
            .Create();

    public static User WithInsurance(this User user, Action<Insurance>? configure = null)
    {
        var insurance = Faker.Build<Insurance>()
            .Without(f => f.Id)
            .With(f => f.User, user)
            .With(f => f.UserId, user.Id)
            .Create();

        configure?.Invoke(insurance);

        user.Insurance = insurance;

        return user;
    }
```

Reply:
I should add that this makes more sense at an integration level, when database constraints become involved. Unit level it might be easier just construct a supply.

Reply:
I updated to follow this more closely. I generate a random number of supplies to add to the List of supplies on the device. I kept the Action Delegate, but it will put the same value on all of the supplies. I don't see a way around that as it's random amount of supplies created.

```
public static Device WithSupplies(this Device device, Action<Supply>? configure = null)
    {
        var supplies = new AutoFaker<Supply>()
            .RuleFor(s => s.Id, f => f.Random.Int(min: 1, max: int.MaxValue))
            .RuleFor(s => s.DeviceId, f => device.Id)
            .RuleFor(s => s.Device, f => device)
            .Generate(new Random().Next(1, 100));

        supplies.ForEach(supply => configure?.Invoke(supply));

        (device.Supplies ??= []).AddRange(supplies);

        return device;
    }
```